bundles/org.openhab.core.config.discovery.usbserial.otheros/src/main/java/org/openhab/core/config/discovery/usbserial/otheros/internal/OtherOSUsbSerialDiscovery.java

```diff
@@ -0,0 +1,221 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.config.discovery.usbserial.otheros.internal;
+
+import java.io.UnsupportedEncodingException;
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.usb.UsbDevice;
+import javax.usb.UsbDeviceDescriptor;
+import javax.usb.UsbDisconnectedException;
+import javax.usb.UsbException;
+import javax.usb.UsbHostManager;
+import javax.usb.UsbHub;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.common.ThreadFactoryBuilder;
+import org.openhab.core.config.discovery.usbserial.UsbSerialDeviceInformation;
+import org.openhab.core.config.discovery.usbserial.UsbSerialDiscovery;
+import org.openhab.core.config.discovery.usbserial.UsbSerialDiscoveryListener;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a {@link UsbSerialDiscovery} implementation component that scans the system for USB devices by means of the
+ * {@link org.usb4java} library implementation of the {@link javax.usb} interface.
+ * <p>
+ * It provides USB coverage on non Linux Operating Systems. Linux is already better covered by the scanners in the
+ * {@link org.openhab.core.config.discovery.usbserial.linuxsysfs} module.
+ * <p>
+ * 
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = UsbSerialDiscovery.class, name = OtherOSUsbSerialDiscovery.SERVICE_NAME)
+public class OtherOSUsbSerialDiscovery implements UsbSerialDiscovery {
+
+    protected static final String SERVICE_NAME = "usb-serial-discovery-other-os";
+
+    private final Logger logger = LoggerFactory.getLogger(OtherOSUsbSerialDiscovery.class);
+    private final Set<UsbSerialDiscoveryListener> discoveryListeners = new CopyOnWriteArraySet<>();
+    private final Duration scanInterval = Duration.ofSeconds(15);
+    private final boolean isLinux;
+    private final @Nullable ScheduledExecutorService scheduler;
+
+    private @Nullable ScheduledFuture<?> scanTask;
+    private Set<UsbSerialDeviceInformation> lastScanResult = new HashSet<>();
+
+    @Activate
+    public OtherOSUsbSerialDiscovery() {
+        isLinux = System.getProperty("os.name", "unknown").toLowerCase().contains("linux");
+        scheduler = isLinux ? null
```

Comment:
Can we not create a mechanism in the service to merge results from different serial discovery services instead, and allow it to run on linux as well?
This may increase the chance of finding all info.

Reply:
> merge results from different serial discovery services instead

Yeah. I had a sleepless night last night and I was coding that in my head at 3am

Reply:
Done